### ISSUE TITLE
don't backfeed RTs of mentions

### ISSUE THREAD

**Issue by snarfed** [snarfed/bridgy#509]
e.g. [this post of adactio's](https://adactio.com/journal/9646#comment23373) has tons of RTs of mentions ([example](https://adactio.com/journal/9646#comment23243)). i thought i'd already handled this: we [explicitly don't fetch mention likes or shares](https://github.com/snarfed/bridgy/blob/master/tasks.py#L145), and we [strip any that granary still gives us](https://github.com/snarfed/bridgy/blob/master/tasks.py#L160).

i'm guessing search returns us RTs as well as their original tweets, so we need to explicitly filter those out in `get_activities` if `fetch_shares` is False. sigh.

**Comment by snarfed** [comment]
i'm also suspicious of [how we look at the `retweet` field to identify RTs vs original tweets](https://github.com/snarfed/granary/blob/master/granary/twitter.py#L249), but i think that's unrelated here.

**Comment by kylewm** [comment]
> i'm also suspicious of how we look at the retweet field to identify RTs vs original tweets

What are you suspcious of? "retweeted" seems to be a boolean that is true iff this status is a retweet.

**Comment by snarfed** [comment]
it's subtle, its actually true iff the authenticating user has RTed the tweet. [docs.](https://dev.twitter.com/overview/api/tweets)

**Comment by kylewm** [comment]
good call!! and of course i didn't notice that because i was looking at my own tweets, dumb.

**Comment by snarfed** [comment]
interesting. i think this was actually also triggering #494: when mentions search found RTs of POSSE tweets first, before the normal timeline fetch did, only the mention version would be backfed, and as a normal post, not as an RT, so it would never show up. so we arguably needed this fix before #494 was fully fixed.

example: https://www.brid.gy/twitter/benborges_ , [propagate log](https://www.brid.gy/log?start_time=1445016366&key=aglzfmJyaWQtZ3lyNQsSCFJlc3BvbnNlIid0YWc6dHdpdHRlci5jb20sMjAxMzo2NTUwMjg5MTI2OTU3MTc4ODgM):
```
Params: UnicodeMultiDict([(u'response_key', u'aglzfmJy...')])
Source: benborges_ (Twitter) benborges_, http://brid-gy.appspot.com/twitter/benborges_
Created by this poll: http://brid-gy.appspot.com/log?start_time=1445006829&key=aglzfmJyaWQtZ3lyFwsSB1R3aXR0ZXIiCmJlbmJvcmdlc18M
Starting Response post tag:twitter.com,2013:655028912695717888 https://twitter.com/SyndicatsWatch/status/655028912695717888
Webmention from https://brid-gy.appspot.com/post/twitter/benborges_/655028448918925313 to https://1-m.cc/2015/superbe-intervention-de-pietro-au-workshop-sur-migration-et-capitalisme
Using cached webmention endpoint for https 1-m.cc: https://1-m.cc/webmention/
Sending...
Starting new HTTPS connection (1): 1-m.cc
"POST /webmention/ HTTP/1.1" 202 None
Sent! {'body': u'', 'request': u'POST https://1-m.cc/webmention/ (with source=https://brid-gy.appspot.com/post/twitter/benborges_/655028448918925313, target=https://1-m.cc/2015/superbe-intervention-de-pietro-au-workshop-sur-migration-et-capitalisme)', 'http_status': 202}
Setting last_webmention_sent
```
